Article.js

```diff
@@ -1,29 +1,107 @@
 export class Article {
   constructor(articleData, app) {
     this.articleData = articleData;
-    this.app = app
+    this.app = app;
     this.articleContainer = null;
     this.editButton = null;
     this.deleteButton = null;
     this.createNewArticle();
+    this.initializeDeleteButton();
+    this.titleElement = null;
+    this.contentElement = null;
+    this.initializeEditButton();
+    this.editTitleInput = null;
+    this.editContentInput = null;
+    this.errorMessageEdit = null;
   }
   createNewArticle() {
     this.articleContainer = document.createElement("div");
     this.articleContainer.classList.add("article-container");
-    const titleElement = document.createElement("h2");
-    const id = this.articleData.id;
-    titleElement.innerText = this.articleData.title;
-    const contentElement = document.createElement("p");
-    contentElement.innerText = this.articleData.content;
+    this.titleElement = document.createElement("h2");
+    this.titleElement.innerText = this.articleData.title;
+    this.contentElement = document.createElement("p");
+    this.contentElement.innerText = this.articleData.content;
     this.editButton = document.createElement("button");
     this.editButton.innerText = "Edit article";
     this.deleteButton = document.createElement("button");
     this.deleteButton.innerText = "Delete article";
-    this.articleContainer.append(titleElement);
-    this.articleContainer.append(contentElement);
+    this.articleContainer.append(this.titleElement);
+    this.articleContainer.append(this.contentElement);
     this.articleContainer.append(this.editButton);
     this.articleContainer.append(this.deleteButton);
     this.app.appWrapper.append(this.articleContainer);
   }
 
+  initializeDeleteButton() {
+    this.deleteButton.addEventListener(
+      "click",
+      this.askServerToDeleteArticleAndRefresh,
+    );
+  }
+  askServerToDeleteArticleAndRefresh = async () => {
+    const deleteResponse = await fetch(
+      `http://localhost:3000/articles/${this.articleData.id}`,
+      {
+        method: "DELETE",
+      },
+    );
+    if (deleteResponse.status === 200) {
+      this.app.refresh();
+    } else {
+      this.app.appWrapper.innerText = "Server error.";
+    }
+  };
+
+  initializeEditButton = () => {
+    this.editButton.addEventListener("click", this.replaceArticleWithEditForm);
+  };
+
+  replaceArticleWithEditForm = () => {
+    this.editForm = document.createElement("form");
+    this.editTitleInput = document.createElement("input");
+    this.editTitleInput.value = this.articleData.title;
+    this.editContentInput = document.createElement("input");
+    this.editContentInput.value = this.articleData.content;
+    const sendEditedArticleButton = document.createElement("button");
```

Comment:
usually when you are doing variable/const declaration you should keep it at the top of the function, just for the clear code purpose. Sometimes it is not possible but this is not the case